src/popup/utils/utils.ts

```diff
@@ -658,6 +670,25 @@ export function isTxOfASupportedType(encodedTx: string, isTxBase64 = false) {
   return SUPPORTED_TX_TYPES.includes(SCHEMA.OBJECT_ID_TX_TYPE[txObject.tag]);
 }
 
+// eslint-disable-next-line no-unused-vars
+export function pipe<T = any[]>(fns: ((data: T) => T)[]) {
+  return (data: T) => fns.reduce((currData, func) => func(currData), data);
+}
+
+export function isTxDex(tx: ITx, dexContractx: IDexContracts) {
+  const { wae, router } = dexContractx;
+
+  return !!(
+    tx?.contractId
+    && tx?.function
+    && (
+      Object.values(FUNCTION_TYPE_DEX).flat()
+        .includes(tx?.function as TxFunctionRaw)
```

Comment:
There's a check for the existence of `tx.function` so ne need to use question mark here.